src/components/lib/FlowGrid/FlowGrid.jsx

```diff
@@ -56,56 +58,87 @@ export default class FlowGrid extends Component{
     }
   }
 
+  _selectedItems() {
+   return this.props.items.filter(i => isWithinRegion(i.location, this.props.selectedRegion))
+  }
+
+  _handleRemoveSelectedItems() {
+    const selectedItems = this._selectedItems()
+    selectedItems.map(i => {this.props.onRemoveItem(i.key)})
+  }
+
   _handleKeyDown(e){
+    if (e.target && e.target.type === 'textarea') { return }
+    if (e.keyCode === 8 || e.keyCode === 46) {
+      this._handleRemoveSelectedItems()
+      e.preventDefault()
+    }
+
     let direction = keycodeToDirection(e.keyCode)
     if (direction) {
       e.preventDefault()
       const size = ({columns: this._columnCount(), rows: this._rowCount()})
-      let newLocation = new DirectionToLocation(size, this.props.selected)[direction]()
+      let newLocation = new DirectionToLocation(size, this.props.selectedCell)[direction]()
       this.props.onSelectItem(newLocation)
-    } else if (e.keyCode == '17' || e.keyCode == '224' || e.keyCode == '91') {
+      this.refs[`cell-${newLocation.row}-${newLocation.column}`].decoratedComponentInstance._focus()
+    } else if (!e.shiftKey && (e.keyCode == '17' || e.keyCode == '224' || e.keyCode == '91')) {
       e.preventDefault()
       this.setState({ctrlPressed: true})
-    } else if (e.keyCode == '86' && this.state.ctrlPressed) {
-      this.props.onPaste()
-    } else if (e.keyCode == '67' && this.state.ctrlPressed) {
-      this.props.onCopy()
+    } else if (this.state.ctrlPressed) {
+      if (e.keyCode == '86') {
+        this.props.onPaste()
+      } else if (e.keyCode == '67') {
+        this.props.onCopy()
+      } else if (e.keyCode == '88') {
+        this.props.onCopy()
+        this._handleRemoveSelectedItems()
+      }
     }
   }
 
-  size(){
+  _handleEndRangeSelect(corner1) {
+    const corner2 = this.props.selectedCell
+
+    if (!isLocation(corner2)) {
+      this.props.onSelectItem(corner1)
+      return
+    }
+
+    this.props.onMultipleSelect(corner1, corner2)
+  }
```

Comment:
Multi-selecting a 1x1 grid is perfectly valid; actually is used often. Right now I believe region select controls most of the styling, so should be present on singleton selections as well (and is presently). I'll add an else statement momentarily.